src/main/java/frc/robot/commands/LevelRobot.java

```diff
@@ -0,0 +1,61 @@
+package frc.robot.commands;
+
+import com.ctre.phoenix.sensors.WPI_Pigeon2;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.ExampleSubsystem;
+import frc.robot.subsystems.Swerve.DrivebaseSubsystem;
+public class LevelRobot extends CommandBase{
+    private final DrivebaseSubsystem drive;
+    private final PIDController pid = new PIDController(
+        Constants.LevelRobotConstants.PID_kP, 
+        Constants.LevelRobotConstants.PID_kI, 
+        Constants.LevelRobotConstants.PID_kD);
+    private final PIDController headingPid = new PIDController(
+        Constants.LevelRobotConstants.HEADING_PID_kP,
+        Constants.LevelRobotConstants.HEADING_PID_kI,
+        Constants.LevelRobotConstants.HEADING_PID_kD);
+    private double startingYaw;
+    
+    public LevelRobot(SwerveSubsystem drivetrain){
+        drive = drivetrain;
+        addRequirements(drive);
+    }
+    // Called when the command is initially scheduled.
+    @Override
+    public void initialize() {
+        pid.setTolerance(0);
+        pid.setSetpoint(0);
+        startingYaw = drive.getYawAsDouble() % 360;
+    }
+
+    // Called every time the scheduler runs while the command is scheduled.
+    @Override
+    public void execute() {
+        double roll = drive.getRoll();
```

Comment:
Shouldnt you be watching pitch? Because that would be the intake moving up and down. Is this a symptom of how our gyro is mounted? But also what happens if the driver, strafes sideways on to the platform? Wouldn't you need to actually PID off roll then? Or is that a case that we just dont touch?

Reply:
hmmmmm
hyro is mounted so that we want to use roll, and i think the answer to going on sideways is "dont do that". And if it's an issue on practice field, we can write a switch for which one to use based on robot heading. I dont think its needed though.